File: Checkout System/SemicolonStore.java
import java.util.Scanner;
import java.util.ArrayList;

public class SemicolonStore{
    public static void main(String[] args){
    
    Scanner input = new Scanner(System.in);
    
    ArrayList<String> items = new ArrayList<>();
    ArrayList<Integer> quantities = new ArrayList<>();
    ArrayList<Double> prices = new ArrayList<>();
    
    System.out.println("Enter Customer's Name");
    String customerName = input.nextLine();
    
    String addMoreItems = "yes";
                                                //SHOOPING LOOP
    while (addMoreItems.equalsIgnoreCase("yes")){
    System.out.println("What did the Customer buy?");
    items.add(input.nextLine());
    
    System.out.println("How many Pieces?");
    quantities.add(input.nextInt());
    input.nextLine();                       
    
    System.out.println("How much per unit?");
    prices.add(input.nextDouble());
    input.nextLine();
    
    System.out.println("Add more items? (Yes/No)");
    addMoreItems = input.nextLine();
    
    }
    System.out.println("Enter cashier's Name:");
    String cashiersName = input.nextLine();
    
    System.out.println("How much Discount will Customer get (in %)? ");
    double discountPercent = input.nextDouble();
    
    //CALCULATIONS
    
    double subTotal = 0;
    
    for(int count = 0; count < items.size(); count = count + 1){
        subTotal += quantities.get(count) * prices.get(count);
    }
    
    // CALCULATE THE TAX AND DISCOUNT 
    double discountAmount = subTotal * (discountPercent / 100);
    double vat = subTotal * 0.175;
    double billTotal = (subTotal - discountAmount) + vat;
    
    //PRINTING INVOICE
    System.out.println("\n--- RECEIPT ---");
    System.out.println("Customer:" + customerName);
    System.out.println("Cashier:" + cashiersName);
    System.out.println("SubTotal:" + subTotal);
    System.out.println("Discount:" + discountAmount);
    System.out.println("Vat @ 17.5%:" + vat);
    System.out.println("BillTotal:" + billTotal);
    
    System.out.println("\nHow much did Customer pay?");
    double amountPaid = input.nextDouble();
    double balance = amountPaid - billTotal;
    
    System.out.println("Balance:" + balance);
    System.out.println("THANK YOU FOR YOUR PATRONAGE");
 
    }
}

File: Checkout System/JavaScript/semicolonStore.js
const prompt = require("prompt-sync")()
    
    let customerName = prompt("Enter Customer's Name: ");
    
    let addMoreItems = "yes";
    
    const items = []
    const quantities = []
    const prices = []
                                                //SHOOPING LOOP
    while (addMoreItems.toLowerCase()==="yes"){
    
        items.push(prompt("what did the customer buy? "));
        
        quantities.push(prompt("How many Pieces? "));
 
        prices.push(prompt("How much per unit? "));
 
        addMoreItems = prompt("Add more items? (Yes/No) ");
    
    }
    
    let cashiersName = prompt("Enter cashier's Name: ")
    
    let discountPercent = prompt("How much Discount will Customer get (in %)? ");
    
    //CALCULATIONS
    
    let subTotal = 0;
    
    for(count = 0; count < items.length; count = count + 1){
        
        subTotal += quantities[count] * prices[count];
    }
    
    // CALCULATE THE TAX AND DISCOUNT 
    let discountAmount = subTotal * (discountPercent / 100);
    let vat = subTotal * 0.175;
    let billTotal = (subTotal - discountAmount) + vat;
    
    //PRINTING INVOICE
   console.log("\n--- RECEIPT ---");
   console.log("Customer:" + customerName);
   console.log("Cashier:" + cashiersName);
   console.log("SubTotal:" + subTotal);
   console.log("Discount:" + discountAmount);
   console.log("Vat @ 17.5%:" + vat);
   console.log("BillTotal:" + billTotal);

    let amountPaid = prompt("\nHow much did Customer pay? ");
    let balance = amountPaid - billTotal;
    
   console.log("Balance:" + balance);
   console.log("THANK YOU FOR YOUR PATRONAGE");
 
 

